crates/tiny/src/main.rs

```diff
@@ -178,3 +193,41 @@ fn run(
 
     runtime.block_on(local);
 }
+
+// Helper to parse SASL config into a libtiny_client SASL struct
+fn sasl_from_config(tls: bool, sasl_config: &config::SASLAuth) -> Result<SASLAuth, String> {
+    match sasl_config {
+        config::SASLAuth::Plain { username, password } => Ok(SASLAuth::Plain {
+            username: username.to_string(),
+            password: password.to_string(),
+        }),
+        config::SASLAuth::External { pem } => {
+            // TLS must be on for EXTERNAL
+            if !tls {
+                Err("TLS not enabled".to_string())
+            } else {
+                // load in a cert and private key for TLS client auth
+                match File::open(pem) {
+                    Ok(file) => {
+                        let mut buf = BufReader::new(file);
+                        // extract certificate
+                        let cert = rustls_pemfile::certs(&mut buf)
+                            .map_err(|e| format!("Could not parse pkcs8 PEM: {}", e))?
```

Comment:
```suggestion
                            .map_err(|e| format!("Could not parse PKCS8 PEM: {}", e))?
```

AFAIU, PKCS is an abbreviation for "Public-Key Cryptography Standards" (https://en.wikipedia.org/wiki/PKCS_8) so this should be in all caps.

Apparently it's also written as "PKCS <span>#</span>8" sometimes. Not sure about "PKCS8" vs. "PKCS <span>#</span>8".